compiler/lib/src/main/scala/analysis/Analyzers/TypeExpressionAnalyzer.scala

```diff
@@ -42,9 +42,9 @@ trait TypeExpressionAnalyzer
     } yield a
   }
 
-  def typeNameNode(a: Analysis, node: AstNode[Ast.TypeName]) = matchTypeNameNode(a, node)
+  def typeNameNode(a: Analysis, node: AstNode[Ast.TypeName]): Out = matchTypeNameNode(a, node)
```

Comment:
I replaced it.

Note that the confusion comes from here:

```
trait AstStateVisitor extends AstVisitor {

  type State

  type In = State

  type Out = Result.Result[State]

  type Result = Result.Result[State]
...
```

Generally, I prefer avoiding type synonyms like `Out` and `Result` because one can easily get confused with inconsistent usage as was the case here.